utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,219 @@
+
+    returns (SUCCESS, comments)
+    SUCCESS means all hist files matched their corresponding baseline
+    """
+    rundir   = case.get_value("RUNDIR")
+    if baseline_dir is None:
+        baselineroot = case.get_value("BASELINE_ROOT")
+        basecmp_dir = os.path.join(baselineroot, case.get_value("BASECMP_CASE"))
+        dirs_to_check = (baselineroot, basecmp_dir)
+    else:
+        basecmp_dir = baseline_dir
+        dirs_to_check = (basecmp_dir,)
+
+    for bdir in dirs_to_check:
+        if not os.path.isdir(bdir):
+            return False, "ERROR %s does not exist" % bdir
+
+    return _compare_hists(case, rundir, basecmp_dir)
+
+def get_extension(model, filepath):
+    """
+    For a hist file for the given model, return what we call the "extension"
+
+    model - The component model
+    filepath - The path of the hist file
+
+    >>> get_extension("cpl", "cpl.hi.nc")
+    'hi'
+    >>> get_extension("cpl", "cpl.h.nc")
+    'h'
+    >>> get_extension("cpl", "cpl.h1.nc")
+    'h1'
+    >>> get_extension("cpl", "TESTRUNDIFF_Mmpi-serial.f19_g16_rx1.A.melvin_gnu.C.fake_testing_only_20160816_164150-20160816_164240.cpl.hi.0.nc.base")
+    'hi'
+    >>> get_extension("cpl", "TESTRUNDIFF_Mmpi-serial.f19_g16_rx1.A.melvin_gnu.C.fake_testing_only_20160816_164150-20160816_164240.cpl.h.nc")
+    'h'
+    """
+    basename = os.path.basename(filepath)
+    ext_regex = re.compile(r'.*%s.*[.](h.?)([.][^.]+)?[.]nc' % model)
+    m = ext_regex.match(basename)
+    expect(m is not None, "Failed to get extension for file '%s'" % filepath)
+    return m.groups()[0]
+
+def generate_baseline(case, baseline_dir=None):
+    """
+    copy the current test output to baseline result
+
+    case - The case containing the hist files to be copied into baselines
+    baseline_dir - Optionally, specify a specific baseline dir, otherwise it will be computed from case config
+
+    returns (SUCCESS, comments)
+    """
+    rundir   = case.get_value("RUNDIR")
+    if baseline_dir is None:
+        baselineroot = case.get_value("BASELINE_ROOT")
+        basegen_dir = os.path.join(baselineroot, case.get_value("BASEGEN_CASE"))
+    else:
+        basegen_dir = baseline_dir
+    testcase = case.get_value("CASE")
+
+    if not os.path.isdir(basegen_dir):
+        os.makedirs(basegen_dir)
+
+    comments = "Generating baselines into '%s'\n" % basegen_dir
+    num_gen = 0
+    for model in _iter_model_file_substrs(case):
+        comments += "  generating for model '%s'\n" % model
+        hists = _get_all_hist_files(testcase, model, rundir)
+        num_gen += len(hists)
+        for hist in hists:
+            ext = get_extension(model, hist)
+            basename = "%s.%s.nc" % (model, ext)
+            baseline = os.path.join(basegen_dir, basename)
+            if os.path.exists(baseline):
+                os.remove(baseline)
+
+            shutil.copy(hist, baseline)
+            comments += "    generating baseline '%s'\n" % baseline
+
+    return True, comments
```

Comment:
Actually, to be consistent with compare, it should probably throw an exception given that something must be seriously wrong if there are no hist files present. Thanks for catching this.